solutions/cses-1202.mdx

```diff
@@ -5,86 +5,99 @@ title: Investigation
 author: Andrew Wang
 ---
 
-**Time Complexity:** $\mathcal{O}(E\log V)$
+## Explanation
+
+We can run Dijkstra keeping track of
+- the distance: $\texttt{dist}[]$
+- the number of ways with the minimum distance: $\texttt{num}[]$
+- the minimum flights with the minimum distance: $\texttt{minf}[]$, and 
+- the maximum flights with the minimum distance: $\texttt{maxf}[]$.
 
-We can run Dijkstra keeping track of the distance: $dist[]$, the number of ways
-with the minimum distance: $num[]$, the minimum flights with the minimum
-distance: $minf[]$, and the maximum flights with the minimum distance: $maxf[]$.
 For every node, $v$, we take into consideration all of its neighbors, $u$. If we
 can reach $u$ in a shorter distance than its current minimum, we update the
-distance and reset $num[u]$, $minf[u]$, and $maxf[u]$. We also have to take into
+distance and reset $\texttt{num}[u]$, $\texttt{minf}[u]$, and $\texttt{maxf}[u]$.
+
+We also have to take into
 consideration if we can reach $u$ in an equivalent distance. If so, we update:
 
 $$
-num[u] = num[v] + num[u]
+\texttt{num}[u] = \texttt{num}[v] + \texttt{num}[u]
 $$
 
 $$
-minf[u] = min(minf[u], minf[v] + 1)
+\texttt{minf}[u] = \min(\texttt{minf}[u], \texttt{minf}[v] + 1)
 $$
 
 $$
-maxf[u] = max(maxf[u], maxf[v] + 1)
+\texttt{maxf}[u] = \max(\texttt{maxf}[u], \texttt{maxf}[v] + 1)
 $$
 
+## Implementation
+
+**Time Complexity:** $\mathcal{O}((N + M)\log N)$
+
+<LanguageSection>
+	
+<CPPSection>
+
 ```cpp
 #include <bits/stdc++.h>
 using namespace std;
 
 typedef long long ll;
-vector<pair<ll, int>> edge[100001];
-ll dist[100001]; ll num[100001];
-int minf[100001];int maxf[100001];
-bool v[100001];
-ll MAX = 1000000000000000L; ll MOD = 1000000007L;
-void djikstra(int s){
+
+const int MAXN = 100001;
+const ll MAX = 0x3f3f3f3f3f3f3f3f;
+const int MOD = 1000000007;
+
+vector<pair<ll, int>> edge[MAXN];
+ll dist[MAXN], num[MAXN];
+int minf[MAXN], maxf[MAXN];
+bool v[MAXN];
+
+void djikstra(int s) {
 	priority_queue<pair<ll, int>, vector<pair<ll, int>>, greater<pair<ll, int>>> pq;
-	pair<ll, int> p; p.first = 0; p.second = s;
-	pq.push(p);
-	num[s] = 1; dist[s] = 0;
-	while(!pq.empty()){
-		pair<ll, int> cur = pq.top(); pq.pop();
-		if(!v[cur.second]){
-			v[cur.second] = true;
-			for(pair<ll, int> e : edge[cur.second]){
-				ll cost = e.first; int next = e.second;
-				ll alt = cost + dist[cur.second];
-				if(alt == dist[next]){
-					num[next] = (num[next] + num[cur.second]) % MOD;
-					minf[next] = min(minf[next], minf[cur.second]+1);
-					maxf[next] = max(maxf[next], maxf[cur.second]+1);
-					e.first = dist[next];
-					pq.push(e);
-				}
-				else if(alt < dist[next]){
-					dist[next] = alt;
-					num[next] = num[cur.second];
-					minf[next] = minf[cur.second]+1;
-					maxf[next] = maxf[cur.second]+1;
-					e.first = dist[next];
-					pq.push(e);
+	pq.push({dist[s] = 0, s});
+	num[s] = 1;
+	while (!pq.empty()) {
+		auto [cost, vert] = pq.top(); pq.pop();
+		if (!v[vert]) {
```

Comment:
if (v[vert]) {
	continue;
}
reduces indentation, making lines shorter & nicer to look at